apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,311 @@
+static void wm_connect_by_rssi_deinit(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
+	}	
+	WT_LEAVE;
+}
+
+static void _wt_print_wifi_ap_profile(wifi_manager_ap_config_s *config, char *title)
+{
+	WT_LOGP(TAG, "====================================\n");
+	if (title) {
+		WT_LOGP(TAG, "%s\n", title);
+	}
+	WT_LOGP(TAG, "------------------------------------\n");
+	WT_LOGP(TAG, "SSID: %s\n", config->ssid);
+	if (config->ap_auth_type == WIFI_MANAGER_AUTH_UNKNOWN
+		|| config->ap_crypto_type == WIFI_MANAGER_CRYPTO_UNKNOWN) {
+		WT_LOGP(TAG, "SECURITY: unknown\n");
+	} else {
+		char security_type[21] = {0,};
+		strncat(security_type, g_wifi_test_auth_method[config->ap_auth_type], 20);
+		wifi_manager_ap_auth_type_e tmp_type = config->ap_auth_type;
+		if (tmp_type == WIFI_MANAGER_AUTH_OPEN
+			|| tmp_type == WIFI_MANAGER_AUTH_IBSS_OPEN
+			|| tmp_type == WIFI_MANAGER_AUTH_WEP_SHARED) {
+			WT_LOGP(TAG, "SECURITY: %s\n", security_type);
+		} else {
+			strncat(security_type, "_", strlen("_"));
+			strncat(security_type, g_wifi_test_crypto_method[config->ap_crypto_type],
+					strlen(g_wifi_test_crypto_method[config->ap_crypto_type]));
+			WT_LOGP(TAG, "SECURITY: %s\n", security_type);
+		}
+	}
+	WT_LOGP(TAG, "====================================\n");
+}
+
+static void _wt_scan(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;	
+	res = wifi_manager_scan_ap(NULL);	
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "scan Fail");
+		return;
+	}
+	WIO_TEST_WAIT; // wait the scan result
+	WT_LEAVE;
+}
+
+static void _wt_connect(void *arg)
+{
+	WT_ENTER;
+	struct wt_options *ap_info = (struct wt_options *)arg;
+	wifi_manager_ap_config_s apconfig;
+	strncpy(apconfig.ssid, ap_info->ssid, WIFIMGR_SSID_LEN);
+	apconfig.ssid_length = strlen(ap_info->ssid);
+	apconfig.ssid[WIFIMGR_SSID_LEN] = '\0';
+	apconfig.ap_auth_type = ap_info->auth_type;
+	if (ap_info->auth_type != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig.passphrase, ap_info->password, WIFIMGR_PASSPHRASE_LEN);
+		apconfig.passphrase[WIFIMGR_PASSPHRASE_LEN] = '\0';
+		apconfig.passphrase_length = strlen(ap_info->password);
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	} else{
+		apconfig.passphrase[0] = '\0';
+		apconfig.passphrase_length = 0;
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	}
+
+	_wt_print_wifi_ap_profile(&apconfig, "Connecting AP Info");
+
+	wifi_manager_result_e res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, "wait join done");
+	WIO_TEST_WAIT;
+	WT_LEAVE;
+}
+
+static void _wt_disconnect(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "disconnect fail (%d)", res);
```

Comment:
Minor suggestion to make the logs more descriptive please.
Could you put in something like "Failed to disconnect <wifi manager entitiy/whatever the name is>, ret : < >"

Reply:
Done wherever required.